internal/plugin/connectors/tcp/mssql/connector.go

```diff
@@ -186,32 +201,40 @@ func (connector *SingleUseConnector) Connect(
 	// Write the prelogin packet back to the user
 	err = mssql.WritePreloginWithPacketType(clientBuffer, preloginResponse, mssql.PackReply)
 	if err != nil {
-		return connector.sendError("failed to write prelogin response: %s", err)
+		wrappedError := errors.Wrap(err, "failed to write prelogin response")
+		connector.sendError(wrappedError)
+		return nil, wrappedError
 	}
 
-	// Just like above, we don't need to concern ourselves with what the client would
-	// typically be sending to the server, since we are handling all communication with
-	// the server ourselves. We read the next packet so the client isn't blocked,
-	// while the driver handles the handshake.
-	err = clientBuffer.ReadNextPacket()
+	// We parse the client's Login packet so that we can pass on params to the server.
+	clientLogin, err := mssql.ReadLogin(clientBuffer)
 	if err != nil {
-		return connector.sendError("Failed to handle client prelogin request: %s", err)
+		wrappedError := errors.Wrap(err, "failed to handle client login")
+		connector.sendError(wrappedError)
+		return nil, wrappedError
 	}
 
-	// TODO: 	Send the login response to the client 	 (#1016)
-	// TODO: 	Verify appropriate errors are passed to the client (#1013)
-
-	// TODO: 	Replace this with an actual 'ok' message from the server
-	//			once login completes within the driver
-	if _, err = clientConn.Write(connector.CreateAuthenticationOKMessage()); err != nil {
-		return connector.sendError("Failed to send a successful authentication"+
-			" response to the client"+
-			": %s", err)
-	}
+	// Send the client login so that secretless can send honest client params to the server
+	clientLoginChannel <- *clientLogin
```

Comment:
I'd like to see a clearer comment here, to call out that this channel will by read inside of the mssql driver, in code that exists in another repo.  Even though I _knew_ this coming in, I still did a double take before realizing what was happening.

Reply:
This is resolved in a following PR -> #1050